### PR TITLE
Adds topic name test cases to topic validation test

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

Adds edge cases to topic name validation test.

Fixes SC-16926

### Type of change

- [ ] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

[Copy acceptance criteria checklist from story for reviewer, or add a brief acceptance criteria here]

[Front-End: add screenshots/videos of changes made]

### Definition of Done

- [ ] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [ ] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [ ] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/db/topics_test.go [hunk 1]
```diff
@@ -64,21 +65,37 @@ func TestTopicValidate(t *testing.T) {
...
+		{strings.Repeat("a", 513), db.ErrTopicNameTooLong},
+		{"otters;", db.ErrInvalidTopicName},
+		{"1otter", db.ErrInvalidTopicName},
+		{"-ott-ers-", db.ErrInvalidTopicName},
+		{"Where?AreTheOtters", db.ErrInvalidTopicName},
+		{"otters-123", nil},
+		{"otters_are_cool.", nil},
+		{"otters-are-cool", nil},
+		{"WeLoveOtters", nil},
+		{"otters-Are.Cool_01", nil},
```
Comment (pdeziel) [inline pkg/tenant/db/topics_test.go:86]: Awesome, thanks for adding all of the test cases 😎

**Review by pdeziel — APPROVED** [review]
Looks good!
